libdrgn/program.c

```diff
@@ -600,7 +603,9 @@ drgn_program_set_core_dump_fd_internal(struct drgn_program *prog, int fd,
 out_notes:
 	// Reset anything we parsed from ELF notes.
 	prog->aarch64_insn_pac_mask = 0;
-	free(prog->vmcoreinfo.raw);
+	// Free vmcoreinfo buffer if it was not provided by the caller
+	if (!had_vmcoreinfo)
+		free(prog->vmcoreinfo.raw);
 	memset(&prog->vmcoreinfo, 0, sizeof(prog->vmcoreinfo));
```

Comment:
Oooh wow my mistake, should have re-read it more carefully. We need to preserve `prog->vmcoreinfo` because it won't get parsed again.